Assets/Scripts/Elements/MoveAtoB.cs

```diff
@@ -1,22 +1,33 @@
 ﻿using UnityEngine;
+using UnityEditor;
 using System.Collections;
 
-public class MoveAtoB : MonoBehaviour {
-
-	private Vector3 pointA;
-	[SerializeField]private Vector3 pointB = new Vector3(2,0,0);
-
-	[Range(0.0f, 20.0f)]public float speed = 2f;
-
-	void Awake () {
-		pointA = new Vector3(transform.position.x, transform.position.y, transform.position.z);
-		pointB = transform.TransformPoint(pointB);
-	}
-
-	void Update() {
-		// based on the speed set movement step
-		float step = speed * Time.deltaTime;
-		// move GameObject from current position (GameObject own position or if defined point A position) to point B 
-		transform.position = Vector3.MoveTowards(transform.position, pointB, step);
-	}
+public class MoveAtoB : MonoBehaviour
+{
+    [SerializeField]
+    private Vector3 pointA;
+    [SerializeField]
+    private Vector3 pointB = new Vector3(2, 0, 0);    
+    [Range(0.0f, 20.0f)]
+    public float speed = 2f;    
+    void Awake()
+    {
+        pointA = new Vector3(transform.position.x, transform.position.y, transform.position.z);
+        pointB = transform.TransformPoint(pointB);
+    }
+    void FixedUpdate()
+    {
+        StartCoroutine(Move());
+    }
+    IEnumerator Move()
+    {
+        for (float f = 1f; f >= 0; f -= 0.1f)
```

Comment:
Por que esse `for`? Pq ele executa só 10 vezes? Como sabe que ja chegou no target pra parar de rodar o Coroutine?

Nao seria melhor um `while` que verifica se a distancia atual pro target é pequena, ou zero?